google-cloud-firestore/src/main/java/com/google/cloud/firestore/Query.java

```diff
@@ -1037,7 +1102,12 @@ public Query endAt(@Nonnull DocumentSnapshot snapshot) {
    * @param responseObserver The observer to be notified when results arrive.
    */
   public void stream(@Nonnull final ApiStreamObserver<DocumentSnapshot> responseObserver) {
-    stream(
+    Preconditions.checkState(
+        !LimitType.Last.equals(Query.this.options.getLimitType()),
+        "Query results for queries that include limitToLast() constraints cannot be streamed. "
+            + "Use Query.get() instead.");
```

Comment:
Can we add some sort of "why" `limitTolast` queries can't be streamed? This statement was very confusing to me when I first read it.

Reply:
I debated adding this to the error message, but I could not come up with something concise. I ended up modifying the JavaDoc for the limitToLast method to say:

```
   * <p>Results for limitToLast() queries are only available once all documents are received. Hence, 
   * limitToLast() queries cannot be streamed via the {@link #stream(ApiStreamObserver)} API.
```

Reply:
That sounds like an okay compromise for the time being, users will be informed about some of the behavior reasons.